File: .repo/system-design-mastery-module-2-database-fundamentals/0-sql-vs-nosql/1-java/src/main/java/academy/starci/sqlvsnosql/sql/SqlController.java
package academy.starci.sqlvsnosql.sql;

import academy.starci.sqlvsnosql.dto.CreateProductDto;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.List;

@RestController
@RequestMapping("/api/sql")
public class SqlController {

    private final SqlService sqlService;

    public SqlController(SqlService sqlService) {
        this.sqlService = sqlService;
    }

    @PostMapping("/products")
    public ResponseEntity<ProductEntity> create(@RequestBody CreateProductDto dto) {
        ProductEntity created = sqlService.create(dto);
        return ResponseEntity.status(HttpStatus.CREATED).body(created);
    }

    @GetMapping("/products")
    public List<ProductEntity> findAll() {
        return sqlService.findAll();
    }

    @GetMapping("/products/category/{cat}")
    public List<ProductEntity> findByCategory(@PathVariable("cat") String category) {
        return sqlService.findByCategory(category);
    }
}

File: .repo/system-design-mastery-module-2-database-fundamentals/0-sql-vs-nosql/2-csharp/Controllers/SqlController.cs
using Microsoft.AspNetCore.Mvc;
using SqlVsNosql.Models;
using SqlVsNosql.Services;
using System.Threading.Tasks;

namespace SqlVsNosql.Controllers
{
    [ApiController]
    [Route("api/sql")]
    public class SqlController : ControllerBase
    {
        private readonly SqlService _sqlService;

        public SqlController(SqlService sqlService)
        {
            _sqlService = sqlService;
        }

        [HttpPost("products")]
        public async Task<IActionResult> Create([FromBody] CreateProductDto dto)
        {
            var created = await _sqlService.CreateAsync(dto);
            return Created($"/api/sql/products/{created.Id}", created);
        }

        [HttpGet("products")]
        public async Task<IActionResult> FindAll()
        {
            var products = await _sqlService.FindAllAsync();
            return Ok(products);
        }

        [HttpGet("products/category/{cat}")]
        public async Task<IActionResult> FindByCategory([FromRoute(Name = "cat")] string category)
        {
            var products = await _sqlService.FindByCategoryAsync(category);
            return Ok(products);
        }
    }
}
